cpp/tests/interpolate/cubic_spline_test.cu

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <time.h>
+#include <sys/time.h>
+#include <string>
+
+#include <gtest/gtest.h>
+#include <utilities/legacy/error_utils.hpp>
+#include <tests/utilities/legacy/cudf_test_utils.cuh>
+#include <tests/utilities/legacy/cudf_test_fixtures.h>
+#include <cudf/column/column_view.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <cuspatial/cubic_spline.hpp>
+#include <cuspatial/utility.hpp>
+
+struct CubicSplineTest : public GdfTest 
+{
+ 
+};
+
+template<typename T>
+auto make_device_column(T* const points, int length) {
+    T *d_p = NULL;
+    RMM_TRY( RMM_ALLOC( &d_p, length * sizeof(T), 0));
+    assert(d_p != NULL);    
+    HANDLE_CUDA_ERROR( cudaMemcpy( d_p, points, length * sizeof(T), cudaMemcpyHostToDevice ) );
+    cudf::column_view col(cudf::data_type{cudf::experimental::type_to_id<T>()}, length, d_p);
+    cudf::column result(col);
+    RMM_FREE(d_p, 0);
+    return result;
+}
+
+auto get_d_expect() {
```

Comment:
I think you should have a host implementation of computing the coefficients, that way you aren't limited to this one single test case. Then to compare, you run the host implementation, create a device column from the output, and compare the columns rather than iterating over elements.

Reply:
This feels like needing to test the implementation, then also needing to test the test of the implementation. I'm supposing that you're worried about the underlying CUDA changing over time, breaking my implementation, while the CPU code remains static? Could I just write a few more fixtures for testing against known/specific inputs?

I don't see how the added engineering of implementing this a second time on host is benefiting my test cases enough to justify it.